tests/framework/Optimizers/GradientDescent/gold/1variable/opt_export_cd_0.csv

```diff
@@ -8,38 +8,38 @@ x,y,ans,stepSize,iteration,accepted,conv_gradient
 2.1431640625,0.5,1.15860859692,0.1220703125,6.0,accepted,0.0
 3.51645507813,0.5,0.420926728435,0.152587890625,7.0,accepted,0.0
 1.79984130859,0.5,2.27310108344,0.190734863281,8.0,rejected,0.0
-3.51645507813,0.5,0.420926728435,0.190734863281,9.0,rerun,0.0
-2.37204589844,0.5,0.622296276884,0.127156575521,10.0,rejected,0.0
-3.51645507813,0.5,0.420926728435,0.127156575521,11.0,rerun,0.0
-2.753515625,0.5,0.0958782696724,0.0847710503472,12.0,accepted,0.0
-3.70718994141,0.5,0.789248108373,0.105963812934,13.0,rejected,0.0
-2.753515625,0.5,0.0958782696724,0.105963812934,14.0,rerun,0.0
-3.3892985026,0.5,0.239170089642,0.070642541956,15.0,rejected,0.0
-2.753515625,0.5,0.0958782696724,0.070642541956,16.0,rerun,0.0
-3.17737087674,0.5,0.049648487802,0.0470950279707,17.0,accepted,0.0
-2.64755181207,0.5,0.196034253796,0.0588687849633,18.0,rejected,0.0
-3.17737087674,0.5,0.049648487802,0.0588687849633,19.0,rerun,0.0
-2.82415816696,0.5,0.0487961777355,0.0392458566422,20.0,accepted,0.0
-3.26567405418,0.5,0.111388328275,0.0490573208028,21.0,rejected,0.0
-2.82415816696,0.5,0.0487961777355,0.0490573208028,22.0,rerun,0.0
-3.11850209177,0.5,0.0221612081439,0.0327048805352,23.0,accepted,0.0
-2.75057218575,0.5,0.0981818388529,0.040881100669,24.0,rejected,0.0
-3.11850209177,0.5,0.0221612081439,0.040881100669,25.0,rerun,0.0
-2.87321548776,0.5,0.0253672744839,0.0272540671127,26.0,rejected,0.0
-3.11850209177,0.5,0.0221612081439,0.0272540671127,27.0,rerun,0.0
-2.9549776891,0.5,0.00319887275602,0.0181693780751,28.0,accepted,0.0
-3.15938319244,0.5,0.0400891125832,0.0227117225939,29.0,rejected,0.0
-2.9549776891,0.5,0.00319887275602,0.0227117225939,30.0,rerun,0.0
-3.09124802466,0.5,0.0131397875381,0.0151411483959,31.0,rejected,0.0
-2.9549776891,0.5,0.00319887275602,0.0151411483959,32.0,rerun,0.0
-3.04582457947,0.5,0.00331389219477,0.0100940989306,33.0,rejected,0.0
-2.9549776891,0.5,0.00319887275602,0.0100940989306,34.0,rerun,0.0
-3.01554228268,0.5,0.000381215900671,0.00672939928708,35.0,accepted,0.0
-2.9398365407,0.5,0.0057122472706,0.00841174910885,36.0,rejected,0.0
-3.01554228268,0.5,0.000381215900671,0.00841174910885,37.0,rerun,0.0
-2.96507178803,0.5,0.00192528092425,0.00560783273923,38.0,rejected,0.0
-3.01554228268,0.5,0.000381215900671,0.00560783273923,39.0,rerun,0.0
-2.98189528625,0.5,0.000517278854294,0.00373855515949,40.0,rejected,0.0
-3.01554228268,0.5,0.000381215900671,0.00373855515949,41.0,rerun,0.0
-2.99311095172,0.5,7.4896212526e-05,0.00249237010632,42.0,accepted,0.0
-2.99311095172,0.5,7.4896212526e-05,0.00249237010632,42.0,final,0.0
+3.51645507813,0.5,0.420926728435,0.190734863281,8.0,rerun,0.0
+2.37204589844,0.5,0.622296276884,0.127156575521,9.0,rejected,0.0
+3.51645507813,0.5,0.420926728435,0.127156575521,9.0,rerun,0.0
+2.753515625,0.5,0.0958782696724,0.0847710503472,10.0,accepted,0.0
+3.70718994141,0.5,0.789248108373,0.105963812934,11.0,rejected,0.0
+2.753515625,0.5,0.0958782696724,0.105963812934,11.0,rerun,0.0
```

Comment:
I see the duplicate step numbers here, I'm struggling to resolve that, unless we want to consider a reject and subsequent rerun as part of the same step, in which case the user input example model run count is only accurate if there are no rejected points.